src/util/address.rs

```diff
@@ -727,6 +727,37 @@ impl Address {
             (Network::Testnet, _) | (Network::Regtest, _) | (Network::Signet, _) => true
         }
     }
+
+    /// Returns true if the given pubkey is directly related to the address payload.
+    ///
+    /// This is determined by directly comparing the address payload with either the
+    /// hash of the given public key or the segwit redeem hash generated from the
+    /// given key. For taproot addresses, the supplied key is assumed to be tweaked
```

Comment:
I think the description of this method - and the return type - may be confused. Since the funciton (and the rest of them) can't detect that the address _is not related_ to the given key. They can only detect certain cases _when it is related_, but not otherwise.

Reply:
I am open for suggestions. I guess your concern is when the function returns `false`? When is this actually an issue? Is it only when an address is derived from scripts as the given key could be referenced in the script?

What would you specifically suggest we change here? Just clarify the doc strings?

Is the issue just with the word "related"? Perhaps something like "derived_from" is better? `is_derived_from_pubkey()`?

Naming is always tough. I have no strong opinion on the name of the function and I'm flexible to change it to any reasonable name. @dr-orlovsky

Reply:
I'm ok with the current name, but also have no strong opinions. @dr-orlovsky I'm skeptical that any user would read `is_related_to` as somehow encompassing exactly the set of all "related" keys, whatever that might mean, as opposed to covering a small set of common useful relations.

Reply:
`is_certainly_related_to_pubkey` may be clearer but quite long...

Reply:
I think the extra length required to be nuanced isn't worth it.

Reply:
Sorry, I was not talking about the method name, but only about documentation and return type

Reply:
@dr-orlovsky Can you be a little more specific about the change that you would like to see? What return type would you desire? How can I clarify the doc strings?

I'm not in disagreement, but I have tunnel vision being the author. I need something a little more actionable to work with.

Reply:
https://github.com/rust-bitcoin/rust-bitcoin/pull/839/files#r836509440

Reply:
IMO this feels like a bit of scope creep as well as a lot of extra code with very little (obvious) benefit. I'm not exactly comfortable making these changes for the reasons that have been stated.